runtime-modules/content-directory/src/lib.rs

```diff
@@ -1689,88 +1858,311 @@ impl<T: Trait> Module<T> {
+        }
+        Ok(())
+    }
+
+    /// Perform all necessary checks to ensure `new_values_for_existing_properties` are valid
+    pub fn ensure_new_property_values_are_valid(
+        entity: &Entity<T>,
+        new_values_for_existing_properties: &ValuesForExistingProperties<T>,
+    ) -> dispatch::Result {
+        for new_value_for_existing_property in new_values_for_existing_properties.values() {
+            let (property, value) = new_value_for_existing_property.unzip();
+
+            // Validate new PropertyValue against the type of this Property and check any additional constraints
+            property.ensure_property_value_to_update_is_valid(
+                value,
+                entity.get_permissions_ref().get_controller(),
+            )?;
+        }
+
+        Ok(())
+    }
+
+    /// Ensure all provided `new_property_values` are already exist in `entity_property_values` map
+    pub fn ensure_all_property_values_are_already_added(
+        entity_property_values: &BTreeMap<PropertyId, PropertyValue<T>>,
+        new_property_values: &BTreeMap<PropertyId, PropertyValue<T>>,
+    ) -> dispatch::Result {
+        ensure!(
+            new_property_values
+                .keys()
+                .all(|key| entity_property_values.contains_key(key)),
+            ERROR_UNKNOWN_ENTITY_PROP_ID
+        );
+        Ok(())
+    }
+
+    /// Ensure `new_values_for_existing_properties` are accessible for actor with given `access_level`
+    pub fn ensure_all_property_values_are_unlocked_from(
+        new_values_for_existing_properties: &ValuesForExistingProperties<T>,
+        access_level: EntityAccessLevel,
+    ) -> dispatch::Result {
+        for value_for_new_property in new_values_for_existing_properties.values() {
+            // Ensure Property is unlocked from Actor with given EntityAccessLevel
+            value_for_new_property
+                .get_property()
+                .ensure_unlocked_from(access_level)?;
+        }
+        Ok(())
+    }
+
+    /// Filter `new_property_values` identical to `entity_property_values`.
+    pub fn try_filter_identical_property_values(
+        entity_property_values: &BTreeMap<PropertyId, PropertyValue<T>>,
+        new_property_values: BTreeMap<PropertyId, PropertyValue<T>>,
+    ) -> BTreeMap<PropertyId, PropertyValue<T>> {
+        new_property_values
+            .into_iter()
+            .filter(|(id, new_property_value)| {
+                matches!(
+                    entity_property_values.get(id),
+                    Some(entity_property_value) if *entity_property_value != *new_property_value
+                )
             })
+            .collect()
+    }
+
+    /// Update `entity_property_values` with `new_property_values`.
+    /// Returns `Some(BTreeMap<PropertyId, PropertyValue<T>>)`,
+    /// if `entity_property_values` have been updated, and `None` otherwise.
+    pub fn make_updated_values(
```

Comment:
Why does this exist, what is it doing?

Reply:
It used to update existing `entity_property_values` with `new_property_values`.
Made an attemt to fix its documentation here https://github.com/Joystream/joystream/pull/669/commits/e300b5b05b783ee112fc394e2bfdb940938c6156#diff-0a165b042aa89a04d896ad36b8a33af5R2150